### ISSUE TITLE
Dockstore search showing unpublished workflows

### ISSUE THREAD

**Issue by Ldcabansay** [ga4gh/dockstore#1941]
### Expected behaviour
Dockstore search shouldn't show unpublished workflows.
### Actual behaviour
[Test workflow](https://github.com/Ldcabansay/expeditions/blob/master/u_of_michigan_aligner.wdl) appears in [search](https://dockstore.org/search?search=expeditions) when unpublished.

### Steps to reproduce the behaviour 
Test workflow was published then unpublished, but afterwards workflow still appeared in dockstore search. Workflow appeared in search both for the publishing user and public. 

### Environment
Chrome- 70.0.3538.77, DockstoreAPI - 1.5.1, DockstoreUI - 2.2.2

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/DOCK-178)
┆Sprint: Seabright Sprint 1 Argo
┆Issue Number: DOCK-178

**Comment by coverbeck** [comment]
I can reproduce this way on a local dev build:

1. Publish a workflow
2. Refresh the workflow -- click Refresh button on the workflow in question.
3. Unpublish the workflow.
4. Go to search page, and search for it, and it is still there.

If you leave out step 2, it works.

Really oddly, I had clicked around trying to reproduce this, and had left one workflow unpublished, but still visible in Search, `coverbeck/expeditions`. Then to see if I had the steps right, I tried with a second workflow. I created it against the same GitHub repo, so I gave it a specific name, ending up with `coverbeck/expeditions/fubar`. After I went through the above steps, not only was `coverbeck/expeditions/fubar` still there even though it was unpublished, but `coverbeck/expeditions` went away, i.e., it had been showing up incorrectly, but then stopped doing so. At least that's what I think happened.
